nuxeo-services/nuxeo-usermapper/src/main/java/org/nuxeo/usermapper/extension/AbstractUserMapper.java

```diff
@@ -149,12 +146,15 @@ protected void updateProfile(CoreSession session, Map<String, Serializable> attr
     @SuppressWarnings({ "unchecked", "rawtypes" })
     protected DocumentModel createPrincipal(Map<String, Serializable> attributes) {
 
-        DocumentModel userModel = userManager.getBareUserModel();
-        userModel.getDataModel(userManager.getUserSchemaName()).setMap((Map) attributes);
-        return userManager.createUser(userModel);
+        DocumentModel userModel = getUserManager().getBareUserModel();
+        userModel.getDataModel(getUserManager().getUserSchemaName()).setMap((Map) attributes);
+        return getUserManager().createUser(userModel);
     }
 
     protected abstract void resolveAttributes(Object userObject, Map<String, Serializable> searchAttributes,
             Map<String, Serializable> userAttributes, Map<String, Serializable> profileAttributes);
 
+    public UserManager getUserManager() {
+        return Framework.getService(UserManager.class);
+    }
```

Comment:
Why these changes ?

Reply:
I should have put these changes in a dfferent commit, but having a default cache generates warnings when trying to register caches but not all components have started. 
Here, before by retreiving the user manager in the constructor, the user directory was initialized too soon.